pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -748,6 +751,19 @@ func (r *RepoSyncReconciler) validateNamespaceName(namespaceName string) error {
 	return nil
 }
 
+func (r *RepoSyncReconciler) validateCACertSecret(ctx context.Context, repoSync *v1beta1.RepoSync) error {
+	caCertSecretRefName := v1beta1.GetSecretName(repoSync.Spec.Git.CACertSecretRef)
+
+	if useCACert(caCertSecretRefName) {
+		caCertSecretRef := client.ObjectKey{Namespace: repoSync.Namespace, Name: caCertSecretRefName}
+		keys := GetSecretKeys(ctx, r.client, caCertSecretRef)
```

Comment:
It looks like `GetSecretKeys` swallows an error and returns `nil` if the `Secret` does not exist. We need to handle the error where the `Secret` does not exist and return a separate error message.

Using `validateSecretExist` seems like the right approach, and using the `Secret` returned by that function to validate that the key is present. Note if we do that we should probably refactor the `NotFound` error message to support both use cases, since [currently it is specific to authentication](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/reconcilermanager/controllers/validate_secret.go#L39).

Note we will also want some additional unit tests for this unhappy path

Reply:
refactored `validateSecretExist` to return generic error and error handling is moved to function caller. Added more unit test wrt NotFound error